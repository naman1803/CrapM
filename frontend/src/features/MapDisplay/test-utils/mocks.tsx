import {
  LakeGeoPoint,
  MapElement,
  MapElementType,
  SewageGeoPoint,
} from '../types/MapElements';

export const mockLakeElements: MapElement[] = [
  {
    id: 1,
    dataType: 'lake',
    isSelectable: true,
    elementType: MapElementType.Marker,
    dataPoint: {
      coordinates: [0, 0],
      id: 1,
      name: 'Grrrr Lake',
      additional_notes: 'This is a lake',
    } as LakeGeoPoint,
  },
  {
    id: 2,
    dataType: 'lake',
    isSelectable: true,
    elementType: MapElementType.Marker,
    dataPoint: {
      coordinates: [0, 0],
      id: 2,
      name: 'Obama Lake',
    } as LakeGeoPoint,
  },
];

export const mockSewageElements: MapElement[] = [
  {
    id: 1,
    dataType: 'sewage',
    isSelectable: true,
    elementType: MapElementType.Circle,
    dataPoint: {
      coordinates: [0, 0],
      id: 1,
      facility_owner: 'Eric',
      radius: 100,
      colour: 'Blue',
    } as SewageGeoPoint,
  },
  {
    id: 2,
    dataType: 'sewage',
    isSelectable: true,
    elementType: MapElementType.Circle,
    dataPoint: {
      coordinates: [0, 0],
      id: 2,
      facility_owner: 'Myron',
      radius: 100,
      colour: 'Red',
    } as SewageGeoPoint,
  },
];
