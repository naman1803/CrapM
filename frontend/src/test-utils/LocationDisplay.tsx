import { useLocation } from 'react-router-dom';
/**test component which displays the current url location.
 * use testid = location-display to get value of current location
 */
const LocationDisplay: React.FC = () => {
  const location = useLocation();

  return <div data-testid='location-display'>{location.pathname}</div>;
};
export default LocationDisplay;
