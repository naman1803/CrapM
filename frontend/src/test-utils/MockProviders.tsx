import { ThemeProvider } from '@mui/material';
import { MemoryRouter } from 'react-router-dom';
import { themeOptions } from '@app/theme';
import React from 'react';

// Wraps all providers for the app for testing
export const TestProviders = ({
  children,
  initialRoute = '/',
}: {
  children?: React.ReactNode;
  initialRoute?: string;
}) => {
  return (
    <ThemeProvider theme={themeOptions}>
      <MemoryRouter initialEntries={[initialRoute]}>{children}</MemoryRouter>
    </ThemeProvider>
  );
};
