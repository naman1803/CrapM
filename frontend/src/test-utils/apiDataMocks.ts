import { LakeData } from '@/types/LakeData';
import {
  SewageCoordData,
  SewageData,
  SewageSeverityData,
} from '@/types/SewageData';

export const mockSewageSeverityElement: SewageSeverityData = {
  id: 1,
  facilityOwner: 'Eric',
  contaminants: {
    brainrot: {
      averageSeverityScore: 1.0,
      overallSeverityLevel: 'Moderate',
      averagePercentExceedance: 100.0,
      records: [
        {
          severityScore: 1.0,
          severityLevel: 'Moderate',
          date: '2021-01-01',
        },
        {
          severityScore: 2.0,
          severityLevel: 'High',
          date: '2021-01-02',
        },
      ],
    },
    takingDumb: {
      averageSeverityScore: 2.0,
      overallSeverityLevel: 'High',
      averagePercentExceedance: 200.0,
      records: [
        {
          severityScore: 0.5,
          severityLevel: 'High',
          date: '2021-01-01',
        },
        {
          severityScore: 2.0,
          severityLevel: 'High',
          date: '2021-01-02',
        },
      ],
    },
  },
};

export const mockLakeApiData: LakeData[] = [
  {
    id: 1,
    lakeName: 'Lake Ontario',
    latitude: 45.2,
    longitude: -78.94,
    additionalNotes: null,
  },
  {
    id: 2,
    lakeName: 'Lake Erie',
    latitude: 46.47,
    longitude: -82.78,
    additionalNotes: null,
  },
];

export const mockSewageApiDataSent = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    latitude: 44.49,
    longitude: -77.69,
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    radius: 22.539892,
    colour: "Green"
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    latitude: 44.68,
    longitude: -79.23,
    contaminant: 'PH - HIGH',
    radius: 22.524761,
    colour: "Green"
  },
];
export const mockSewageApiData: SewageCoordData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    latitude: 44.49,
    longitude: -77.69,
    radius: 22.539892,
    colour: "Green"
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    latitude: 44.68,
    longitude: -79.23,
    radius: 22.524761,
    colour: "Green"
  },
];

export const mockSewageSeverityApiData: SewageSeverityData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    contaminants: {
      'BIOCHEMICAL OXYGEN DEMAND': {
        averageSeverityScore: 0.77,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 177.0,
        records: [
          {
            severityScore: 0.77,
            severityLevel: 'Low',
            date: '2016-08-18',
          },
        ],
      },
    },
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    contaminants: {
      'PH - HIGH': {
        averageSeverityScore: 0.95,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 195.4,
        records: [
          {
            severityScore: 0.95,
            severityLevel: 'Low',
            date: '2016-08-30',
          },
        ],
      },
      'BIOCHEMICAL OXYGEN DEMAND': {
        averageSeverityScore: 0.58,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 157.8,
        records: [
          {
            severityScore: 0.58,
            severityLevel: 'Low',
            date: '2016-08-23',
          },
        ],
      },
    },
  },
];

export const mockSewageAllApiData: SewageData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    siteAddress: '123 Mock St',
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-18T00:00:00',
    exceedanceEndDate: '2016-08-18T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 88.5,
    quantityMinimum: null,
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    siteAddress: '1234 Mock St',
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-23T00:00:00',
    exceedanceEndDate: '2016-08-23T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 78.9,
    quantityMinimum: null,
  },
  {
    id: 3,
    facilityOwner: 'Mock Facility 2',
    siteAddress: '1234 Mock St',
    contaminant: 'PH - HIGH',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-30T00:00:00',
    exceedanceEndDate: '2016-08-30T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 97.7,
    quantityMinimum: null,
  },
];
