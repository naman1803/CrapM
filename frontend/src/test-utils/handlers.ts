import { http, HttpResponse } from 'msw';
import {
  mockLakeApiData,
  mockSewageAllApiData,
  mockSewageApiDataSent,
  mockSewageSeverityApiData,
} from './apiDataMocks';

const handlers = [
  http.get('/api/maps/lakes/all', () => {
    return HttpResponse.json(mockLakeApiData);
  }),
  http.get('/api/maps/sewage/all', () => {
    return HttpResponse.json(mockSewageApiDataSent);
  }),
  http.get('/api/data/sewage/all', () => {
    return HttpResponse.json(mockSewageAllApiData);
  }),
  ...mockSewageSeverityApiData.map((severityData) =>
    http.get(`/api/data/sewage/severity/${severityData.id}`, () => {
      return HttpResponse.json(severityData);
    }),
  ),
];
export default handlers;
