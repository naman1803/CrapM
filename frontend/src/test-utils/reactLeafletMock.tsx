import { LeafletEventHandlerFnMap, LatLngExpression } from 'leaflet';

jest.mock('react-leaflet', () => ({
  __esModule: true,
  ...jest.requireActual('leaflet'),
  useMap: () => ({ flyTo: jest.fn() }),
  MapContainer: jest.fn().mockImplementation(({ children }) => (
    <div data-testid='leaflet-map-container'>
      <p>Map Container</p>
      {children}
    </div>
  )),
  Circle: jest
    .fn()
    .mockImplementation(
      ({
        center,
        radius,
        eventHandlers,
        children,
        key,
      }: {
        center: LatLngExpression;
        radius: number;
        eventHandlers: LeafletEventHandlerFnMap;
        children: React.ReactNode;
        key: any;
      }) => (
        <div key={key} data-testid={`Circle-${center.toString()}`}>
          <p>{`Circle-${center.toString()}-radius-${radius}`}</p>
          {Object.entries(eventHandlers).map(([keyE, value]) => (
            <button
              key={keyE}
              onClick={value}
              data-testid={`circle-${center.toString()}-button-${keyE}`}
            >
              {`Circle-button-${keyE}-${center.toString()}`}
            </button>
          ))}
          {children}
        </div>
      ),
    ),
  Marker: jest
    .fn()
    .mockImplementation(
      ({
        position,
        children,
        key,
        eventHandlers,
      }: {
        eventHandlers: LeafletEventHandlerFnMap;
        position: LatLngExpression;
        key: any;
        children: React.ReactNode;
      }) => (
        <div key={key} data-testid={`Marker-${position.toString()}`}>
          <p>{`Marker-${position.toString()}`}</p>
          {Object.entries(eventHandlers).map(([keyE, value]) => (
            <button
              key={keyE}
              onClick={value}
              data-testid={`marker-${position.toString()}-button-${keyE}`}
            >
              {`button-${keyE}-${position.toString()}`}
            </button>
          ))}
          {children}
        </div>
      ),
    ),
  Popup: jest.fn().mockImplementation(({ children }) => (
    <div data-testid='leaflet-popup'>
      <p>Popup</p>
      {children}
    </div>
  )),
  TileLayer: jest.fn().mockImplementation(({ children }) => (
    <div data-testid='leaflet-tilelayer'>
      <p>TileLayer</p>
      {children}
    </div>
  )),
}));
