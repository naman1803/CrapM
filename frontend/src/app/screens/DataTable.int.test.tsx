import '@/test-utils/reactLeafletMock';

import { TestProviders } from '@/test-utils/MockProviders';
import AppRoutes from '../routes';
import LocationDisplay from '@/test-utils/LocationDisplay';
import { render, waitFor } from '@testing-library/react';

describe('DataTable', () => {
  it('renders the data table loading', () => {
    const { getByText } = renderViewPage('/datatable');
    expect(getByText('Sewage Data Overview')).toBeInTheDocument();
    expect(getByText('Loading...')).toBeInTheDocument();
  });
  it('renders the data table with the data', async () => {
    const { getByText, asFragment } = renderViewPage('/datatable');
    expect(getByText('Sewage Data Overview')).toBeInTheDocument();
    expect(getByText('Loading...')).toBeInTheDocument();
    await waitFor(() => {
      expect(getByText('Facility Owner')).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});

const renderViewPage = (initialRoute: string) =>
  render(
    <TestProviders initialRoute={initialRoute}>
      <AppRoutes />
      <LocationDisplay />
    </TestProviders>,
  );
