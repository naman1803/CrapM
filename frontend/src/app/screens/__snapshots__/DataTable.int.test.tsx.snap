// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable renders the data table with the data 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
  >
    <h1
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1it4a80-MuiTypography-root"
    >
      Sewage Data Overview
    </h1>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1it4a80-MuiTypography-root"
      >
        Industrial Sewage Data Overview
      </h1>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-m0q9o2-MuiPaper-root-MuiTableContainer-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12); --Paper-overlay: linear-gradient(rgba(255, 255, 255, 0.051), rgba(255, 255, 255, 0.051));"
      >
        <table
          class="MuiTable-root css-15y9i9h-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-2o0037-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                ID
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                Facility Owner
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                Site Address
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                Contaminant
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                Type of Exceedance
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                Exceedance Start Date
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                Exceedance End Date
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                Contaminant Limit
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                Contaminant Unit
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                Limit Frequency
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                No. of Exceedances
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                Quantity Maximum
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-nlw0ug-MuiTableCell-root"
                scope="col"
              >
                Quantity Minimum
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </div>
  <div
    data-testid="location-display"
  >
    /datatable
  </div>
</DocumentFragment>
`;
